### ISSUE TITLE
Possible false positive for quote-props rule

### ISSUE THREAD

**Issue by Bnaya** [eslint/eslint#5532]
**What version of ESLint are you using?**
v2.3

**What configuration and parser (Espree, Babel-ESLint, etc.) are you using?**
the built in one

**What did you do? Please include the actual source code causing the issue.**
```
const a = {
  'some//key': 1,
};

a.toString();
```
**What did you expect to happen?**
To have no lint error

**What actually happened? Please include the actual, raw output from ESLint.**
```
Unnecessarily quoted property 'some//key' found. (quote-props)
```

I'm using this preset:
https://github.com/airbnb/javascript/blob/master/packages/eslint-config-airbnb/base.js
This is the rule line
https://github.com/airbnb/javascript/blob/master/packages/eslint-config-airbnb/rules/style.js#L114
```
'quote-props': [2, 'as-needed', { 'keywords': false, 'unnecessary': true, 'numbers': false }],
```

**Comment by mysticatea** [comment]
I confirmed it.

I guess `espree.tokenize` ignores the `//key` part: https://github.com/eslint/eslint/blob/master/lib/rules/quote-props.js#L71

**Comment by BYK** [comment]
Looking into this

**Comment by Bnaya** [comment]
@BYK Thanks!
